### PR TITLE
Balanced bin tree

### LINKED ISSUE #430: [Feature Request] <Title of your feature>to see if a binary tree is balanced or not in python and java
### Is your feature request related to a problem? Please describe.
Ab addition to the problems to check if a tree is balanced or not

### Describe the solution you'd like
This will show if a tree is balanced or not
---

**Would you like to work on this feature?**  
- [ ] Yes

### PR DESCRIPTION
**Pull request by Rashigera**
### Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
added program in java and python to check if a tree is balanced or not . its a important concept and is asked in interviews a lot of time by many companies

Fixes #430 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [T ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [ T] My code follows the style guidelines of this project
- [ T] I have performed a self-review of my code
- [ T] I have commented my code, particularly in hard-to-understand areas
- [ T] I have added tests that prove my fix is effective or that my feature works
- [ T] New and existing unit tests pass locally with my changes
- [T ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
We need only `.md` files for our documentation. If you have written your code in another file format, please convert it to `.md` format.

**Comment by Rashigera** [comment]
added only .md files.

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
Please delete the .java and .py files. For the .md file, please read our guide at https://github.com/ajay-dhangar/algo?tab=readme-ov-file#resources-for-guidance.

**Comment by ajay-dhangar** [comment]
> Please delete the .java and .py files. For the .md file, please read our guide at https://github.com/ajay-dhangar/algo?tab=readme-ov-file#resources-for-guidance.

## Resources for Guidance
Here are some resources that may be helpful as you contribute to Algo:
- [Docusaurus Documentation](https://docusaurus.io/docs/docs-introduction)
- [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [Markdown Guide](https://www.markdownguide.org/)
- [MDX Documentation](https://mdxjs.com/docs/)
- [Mermaid Documentation](https://mermaid.js.org/)

**Comment by Rashigera** [comment]
@ajay-dhangar  the correction is done , kindly check and let me know

File: docs/balancedBinTree/balancedBinTreejava.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# Balanced Binary Tree Checker
```
Comment (ajay-dhangar) [inline docs/balancedBinTree/balancedBinTreejava.md:10]: We need on top

```md
---
id: ___
title: ___
sidebar_label: ___
sidebar_position: ___
description: ___
tags: [___, __, ]
---
```

### Detailed Guidelines

- id: Unique identifier for the route.
- title: Title of the problem.
- sidebar_label: Problem number or label for the sidebar.
- tags: Relevant tags for the problem.
- description: A short SEO-friendly description. No HTML tags.

File: docs/balancedBinTree/balancedBinTreejava.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        Solution solution = new Solution();
+
+        // Checking if the tree is balanced
+        if (solution.isBalanced(root)) {
+            System.out.println("The tree is balanced.");
+        } else {
+            System.out.println("The tree is not balanced.");
+        }
+    }
+}
```
Comment (ajay-dhangar) [inline docs/balancedBinTree/balancedBinTreejava.md:97]: And please close your code snippets with ```` ``` ````

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by Rashigera** [comment]
@ajay-dhangar I have made the necessary changes. kindly check and let me know if anymore update or correction is required

File: docs/balancedBinTree/balancedBinTree.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
+---
+id: balanced-binary-tree-checker-python
+title: Balanced Binary Tree Checker in Python
+sidebar_label: 4. Balanced Binary Tree (Python)
```
Comment (ajay-dhangar) [inline docs/balancedBinTree/balancedBinTree.md]: replace `sidebar_label: 4. Balanced Binary Tree (Python)` with `sidebar_label: "Balanced Binary Tree (Python)"`

File: docs/balancedBinTree/balancedBinTree.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+ root.left.right.right.right = Node(7)
+
+# Creating an instance of the Solution class
+ solution = Solution()
+
+# Checking if the tree is balanced
+ if solution.isBalanced(root):
+   print("The tree is balanced.")
+ else:
+    print("The tree is not balanced.")
```
Comment (ajay-dhangar) [inline docs/balancedBinTree/balancedBinTree.md]: close your example with ```` ``` ````

File: docs/balancedBinTree/balancedBinTreejava.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
+---
+id: balanced-binary-tree-checker
+title: Balanced Binary Tree Checker
+sidebar_label: 110. Balanced Binary Tree
```
Comment (ajay-dhangar) [inline docs/balancedBinTree/balancedBinTreejava.md]: replace `sidebar_label: 110. Balanced Binary Tree` with `sidebar_label: Balanced Binary Tree`

File: docs/balancedBinTree/balancedBinTreejava.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+---
+
+# Balanced Binary Tree Checker
+
+## Problem Statement
+
+Given a binary tree, determine if it is height-balanced. A height-balanced binary tree is defined as:
+
+> A binary tree in which the left and right subtrees of every node differ in height by no more than one.
+
```
Comment (ajay-dhangar) [inline docs/balancedBinTree/balancedBinTreejava.md]: you can try 

replace 

```md
> A binary tree in which the left and right subtrees of every node differ in height by no more than one.
```
with 

```md
:::note
A binary tree in which the left and right subtrees of every node differ in height by no more than one.
:::
```

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**